Extension/src/main.ts

```diff
@@ -75,38 +75,71 @@ export async function activate(context: vscode.ExtensionContext): Promise<CppToo
 
     await processRuntimeDependencies();
 
-    const installedPlatform: string | undefined = util.getInstalledBinaryPlatform();
+    const promptForMacArchictureMismatch: PersistentState<boolean> = new PersistentState<boolean>("CPP.promptForMacArchictureMismatch", true);
+
+    // Read archictures of binaries from install.lock
+    const fileContents: string = await util.readFileText(util.getInstallLockPath());
+    let installedPlatformAndArchitecture: util.InstallLockContents;
+    // Just in case we're debugging with an existing install.lock that is empty, assume current platform if empty.
+    if (fileContents.length === 0) {
+        installedPlatformAndArchitecture = {
+            platform: process.platform,
+            architecture: arch
+        };
+    } else {
+        installedPlatformAndArchitecture = <util.InstallLockContents>JSON.parse(fileContents);
```

Comment:
You should check that the properties exist instead of asserting that they are there.  People can mess with files and invalidate your assumptions.

Reply:
The values would equal undefined (or whatever they put in there), and fail to match.  The message might show incorrectly if we don't understand the platform type, but I don't think that is terrible if they are mucking with our data.  We don't really have a failure path here, other than to throw an exception.